### ISSUE TITLE
OCaml 4.03 compatibility

### ISSUE THREAD

**Issue by hcarty** [ocsigen/lwt#220]
The ppx extension doesn't compile with 4.03.0+beta1.

**Comment by Drup** [comment]
I have a patch here: https://github.com/ocsigen/lwt/tree/4.03.0
You need the dev version of ppx_tools.

**Comment by hcarty** [comment]
Trying to `opam pin add lwt -kgit https://github.com/ocsigen/lwt.git#4.03.0` gave me this error:

```
# File "myocamlbuild.ml", line 767, characters 69-98:
# Error: This expression has type Ocamlbuild_plugin.Pathname.t lazy_t
#        but an expression was expected of type string
# Command exited with code 2.
```

**Comment by hcarty** [comment]
Looks like that may be an oasis issue...

**Comment by dbuenzli** [comment]
Would it be please possible to fix this so that we can test 4.03 well ? This is unfortunately a prequisite of utop and some people tricked me into using it.

**Comment by neolem** [comment]
I want to second @dbuenzli . I am using 4.03 beta and am unable to use `utop` since it has requirements on `lwt`.

**Comment by vasilisp** [comment]
@Drup has a workaround for the OASIS issue: https://github.com/Drup/lwt/tree/4.03.0-no-oasis . You can pin that.

@Drup, we now have `ppx_tools` for 4.03, so I guess we should merge your PPX fix. Hopefully the OASIS issue will get fixed in OASIS.

**Comment by Drup** [comment]
Sorry everyone, I did this some time ago and forgot to follow on. This is now done in https://github.com/ocsigen/lwt/pull/227

As @vasilisp said, while waiting for a fix on oasis or an lwt release, please use my version with oasis already generated.

```
opam pin add lwt https://github.com/Drup/lwt.git#4.03.0-no-oasis
```

**Comment by neolem** [comment]
@Drup still failing on `4.03.0+beta1+flambda`. Error below.

`gcc: error: src/unix/jobs-unix/lwt_unix_job_truncate.o: No such file or director`

**Comment by Drup** [comment]
@neolem I just tried on flambda, I can't reproduce, everything compiles fine.

**Comment by neolem** [comment]
@Drup It installs okay now after removing and re-adding the `4.03.0+beta1+flambda. Thanks.

**Comment by avsm** [comment]
Any ETA on an Lwt release with this fix in? 4.03 is around the corner.

**Comment by whitequark** [comment]
4.03 is released and lwt is still broken...

**Comment by Drup** [comment]
@whitequark lwt.2.5.2 is released too, and it's 4.03-compatible.
